### PR TITLE
upload/download + refactoring + splitting into modules

### PR DESCRIPTION
**Pull request by yuriykuzin**
- upload / download without Crowdin CLI
- refactoring, splitting into modules
- tests

### CODE REVIEW

**Review by KashaevaD — APPROVED** [review]

File: lib/config.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  requiredConfigProperties.forEach(property => {
+    if (!crowdinHelperJson[property]) {
+      console.log(`Crowdin: ${COLORS.RED}Error: ${ property } is missing in crowdin-helper.json${COLORS.WHITE}`);
+      isSomeRequiredFieldsAbsent = true;
+    }
+  });
+
+  if (isSomeRequiredFieldsAbsent) {
+    throw new Error();
```
Comment (sfate) [inline lib/config.js]: Do we need this here?
Why can't we just raise an error at this line: https://github.com/yuriykuzin/crowdin-helper/pull/3/files#diff-f7bf2b665273dfa66ffa4ad7c0a52bb9R23

**Review by bolelamx — APPROVED** [review]
looks good :+1:

File: lib/config.js (hunk 1 above)
Comment (yuriykuzin) [reply lib/config.js]: @sfate If we raise an error immediately, we'll tell about only one missed required prop. In current solution we're telling about all missed fields

File: test/crowdin-fetch.spec.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        "status": "skipped"
+    }
+  };
+
+  beforeEach(() => {
+    fetch.resetMocks();
+  });
+
+  test('crowdinFetch calls node-fetch, sends FormData and gets sample JSON response', async () => {
+    fetch.mockResponse(JSON.stringify(sampleResponse));
```
Comment (sfate) [inline test/crowdin-fetch.spec.js]: Should it be `mockResponseOnce`?

File: test/crowdin-fetch.spec.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    const response = await crowdinFetch('export');
+
+    expect(fetch.mock.calls[0][0]).toEqual('https://api.crowdin.com/api/project/my-project-name/export');
```
Comment (sfate) [inline test/crowdin-fetch.spec.js]: What about calling fetch mock once?
e.g.
```js
const request = fetch.mock.calls[0];

expect(request[0]).toEqual('https://api.crowdin.com/api/project/my-project-name/export');
expect(request[1].method).toBe('POST');
//...
```

File: test/crowdin-fetch.spec.js (hunk 2 above)
Comment (yuriykuzin) [reply test/crowdin-fetch.spec.js]: indeed, also moved it to beforeEach section

File: test/crowdin-fetch.spec.js (hunk 3 above)
Comment (yuriykuzin) [reply test/crowdin-fetch.spec.js]: sure!

File: lib/colors.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+const COLORS = {
```
Comment (stevermeister) [inline lib/colors.js]: not sure that it should be a separate file

File: lib/config.js (hunk 1 above)
Comment (stevermeister) [reply lib/config.js]: let user fix it one by one

**Review by sfate — APPROVED** [review]
LGTM 🚒

**Pull request merged**